NineChronicles.Headless/GraphTypes/States/ShopStateType.cs

```diff
@@ -13,7 +17,45 @@ public ShopStateType()
                 resolve: context => context.Source.address);
             Field<NonNullGraphType<ListGraphType<ShopItemType>>>(
                 nameof(ShopState.Products),
-                resolve: context => context.Source.Products.Values
+                arguments: new QueryArguments(
+                    new QueryArgument<IntGraphType>
+                    {
+                        Name = "id",
+                        Description = "Filter for item id."
+                    },
+                    new QueryArgument<ItemSubTypeEnumType>
+                    {
+                        Name = "itemSubType",
+                        Description = "Filter for ItemSubType. see from https://github.com/planetarium/lib9c/blob/main/Lib9c/Model/Item/ItemType.cs#L13"
+                    },
+                    new QueryArgument<IntGraphType>
+                    {
+                        Name = "maximumPrice",
+                        Description = "Filter for item maximum price."
+                    }),
+                resolve: context =>
+                {
+                    var products = context.Source.Products.Values;
```

Comment:
IMHO, it would be better we specify the type of these variables (without `new` or literals) because it can be more readable during code review.